common/cache/domainCache_test.go

```diff
@@ -608,6 +608,453 @@ func (s *domainCacheSuite) TestGetTriggerListAndUpdateCache_ConcurrentAccess() {
+		// and then release the lock
+		s.domainCache.refreshLock.Lock()
+		defer s.domainCache.refreshLock.Unlock()
+		time.Sleep(200 * time.Millisecond)
+		s.domainCache.cacheByID.Store(testCache)
+		s.domainCache.cacheNameToID.Store(testCache)
+	}()
+
+	s.metadataMgr.On("GetDomain", mock.Anything, &persistence.GetDomainRequest{Name: domainName, ID: ""}).Return(nil, nil).Once()
+
+	entry, err := s.domainCache.getDomain(domainName)
+
+	s.Nil(err)
+	// because the code makes deep copies, it's not possible to compare all the pointers directly
+	s.Equal(domainEntry.info.Name, entry.info.Name)
+	s.Equal(domainEntry.info.ID, entry.info.ID)
+	s.Equal(1, len(entry.info.Data))
+	s.Equal("v1", entry.info.Data["k1"])
+}
+
+func (s *domainCacheSuite) Test_getDomainByID_refreshDomainsLockedError() {
+	domainID := "testDomainID"
+
+	s.metadataMgr.On("GetDomain", mock.Anything, &persistence.GetDomainRequest{Name: "", ID: domainID}).Return(nil, nil).Once()
+	s.metadataMgr.On("GetMetadata", mock.Anything).Return(nil, assert.AnError).Once()
+
+	entry, err := s.domainCache.getDomainByID(domainID, false)
+
+	s.Nil(entry)
+	s.ErrorIs(err, assert.AnError)
+}
+
+func (s *domainCacheSuite) Test_refreshLoop_domainCacheRefreshedError() {
+	mockedTimeSource := clock.NewMockedTimeSource()
+
+	s.domainCache.timeSource = mockedTimeSource
+
+	s.metadataMgr.On("GetMetadata", mock.Anything).Return(nil, assert.AnError).Twice()
+
+	go func() {
+		mockedTimeSource.BlockUntil(1)
+		mockedTimeSource.Advance(DomainCacheRefreshInterval)
+		mockedTimeSource.BlockUntil(2)
+		mockedTimeSource.Advance(DomainCacheRefreshFailureRetryInterval)
+		s.domainCache.shutdownChan <- struct{}{}
+	}()
+
+	s.domainCache.refreshLoop()
+}
+
+func (s *domainCacheSuite) Test_refreshDomainsLocked_IntervalTooShort() {
+	mockedTimeSource := clock.NewMockedTimeSource()
+
+	s.domainCache.timeSource = mockedTimeSource
+
+	s.domainCache.lastRefreshTime = mockedTimeSource.Now()
+
+	err := s.domainCache.refreshDomainsLocked()
+	s.NoError(err)
+}
+
+func (s *domainCacheSuite) Test_refreshDomainsLocked_ListDomainsError() {
+	s.metadataMgr.On("GetMetadata", mock.Anything).Return(&persistence.GetMetadataResponse{NotificationVersion: 0}, nil).Once()
+	s.metadataMgr.On("ListDomains", mock.Anything, mock.Anything).Return(nil, assert.AnError).Once()
+
+	err := s.domainCache.refreshDomainsLocked()
+	s.ErrorIs(err, assert.AnError)
+}
+
+func (s *domainCacheSuite) TestDomainCacheEntry_Getters() {
+	info := &persistence.DomainInfo{
+		ID:   uuid.New(),
+		Name: "testDomain",
+	}
+
+	config := &persistence.DomainConfig{
+		Retention: 1,
+	}
+
+	replicationConfig := &persistence.DomainReplicationConfig{
+		ActiveClusterName: cluster.TestCurrentClusterName,
+	}
+
+	configVersion := int64(5)
+
+	failoverNotificationVersion := int64(10)
+
+	notificationVersion := int64(20)
+
+	failoverVersion := int64(30)
+
+	previousFailoverVersion := int64(29)
+
+	failoverEndTime := common.Int64Ptr(40)
+
+	entry := &DomainCacheEntry{
```

Comment:
In general, this is hard to ensure it'll stay up to date. If you need, you can use `common/testing/testdatagen` or the package it uses directly (`github.com/google/gofuzz`) to generate test data which will fill out the entire struct. Then after storing it or whatever and getting it back, just ensuring it's equal to the original data (taking care about deep-copying i guess) should ensure that this will evolve. 

this *may not work* here where there's lots of private structs, atomics etc. so this is general and maybe not accurate in all cases